### ISSUE TITLE
`bundler/setup.rb` script generates by `bundle install --standalone` uses absolute paths for `:path` gem dependencies.

### ISSUE THREAD

**Issue by prakhunov** [rubygems/rubygems#5317]
### Describe the problem as clearly as you can

We package our Ruby applications in a Docker image that gets deployed but since we use a bunch of private git repos, and due to it being a monorepo we also use gems installed via `:path`.

On bundler 2.2.29 (JRuby 9.3 default) ,and even updating to bundler 2.3.6, the `bundler/setup.rb` script has absolute paths to the gems that are installed via `:path`. For additional info `:git` doesn't have this issue. This of course breaks in the resulting docker image because those paths don't exist.

On bundler 2.1.4 this was working fine. 

### Did you try upgrading rubygems & bundler?
Yes I tried 2.3.6.

### Post steps to reproduce the problem

1. Create a project with a Gemfile similar to the following:

```ruby
  gem "dep_a", :path => "../dep_a" # relative path for dep_a
```

2. Run the following:

```bash
bundle package --all
bundle install --standalone --deployment --local
```

The resulting `vendor/cache/bundler/setup.rb` will use the absolute path to `dep_a` which breaks if this is copied to another system or docker.

### Which command did you run?

`bundle install --standalone`

### What were you expecting to happen?
If I have relative paths in my Gemfile, I expected the paths to look like this (as they did in 2.1.4)

```ruby
$:.unshift "#{path}/../../cache/dep_a
```

instead it looks like:

```
$:.unshift "/home/username/main_project/vendor/cache/dep_a"
```

### What actually happened?

The result in the file looks like:

```
/home/username/project/vendor/cache/dep_a
```

### If not included with the output of your command, run `bundle env` and paste the output below

The only difference is during the `bunde package` phase it says the dependencies are using a relative path,
and during the `bundle install --standalone` phase it uses the absolute path in the output.

**Comment by deivid-rodriguez** [comment]
I actually have a vague recollection that this was done intentionally but I can't recall why now. Could you bisect the specific pull request that caused this? Thanks.

**Comment by prakhunov** [comment]
@deivid-rodriguez 

I have not done a bisect, but I'm pretty sure it is this:

https://github.com/rubygems/rubygems/issues/3505

Good old: https://www.hyrumslaw.com/ at work.

**Comment by deivid-rodriguez** [comment]
I think the issue was actually introduced by https://github.com/rubygems/rubygems/pull/4792 and I believe option two makes sense, actually, the original report was exactly that, passing an absolute path as the `path:` value.

**Comment by prakhunov** [comment]
> I think the issue was actually introduced by #4792 and I believe option two makes sense, actually, the original report was exactly that, passing an absolute path as the `path:` value.

I don't think that is the root cause just because it looks like those commits first appear in 2.3.0, and I am running into this issue on 2.2.29 as well:

```bash
git tag --contains d3bba936f05d0f2dc307c1f282064704f0724190
bundler-v2.3.0
bundler-v2.3.1
bundler-v2.3.2
bundler-v2.3.3
bundler-v2.3.4
bundler-v2.3.5
bundler-v2.3.6
v3.3.0
v3.3.1
v3.3.2
v3.3.3
v3.3.4
v3.3.5
v3.3.6
```

Unless it got backported and has a different commit.

**Comment by deivid-rodriguez** [comment]
Yes, it is the root cause. That fix was released with 2.2.26, that's why you're getting the issue on 2.2.29. Try downgrading to 2.2.25 and the issue will disappear. 

> Unless it got backported and has a different commit.

Yep, this is the reason why you got this wrong.

**Comment by prakhunov** [comment]
Makes sense, I'll hopefully take a look this week and see if I can get a patch going.
